### ISSUE TITLE
Scroll-bars appear always on table in IE

### ISSUE THREAD

**Issue by supreethmr** [cerner/terra-core#3468]
# Bug Report
Vertical Scroll bars are displayed always on table even when there is no content to scroll.

## Description

Terra-Table has `overflow-y:scroll` style property for both table header and body when fill prop is set to true on table. 
`overflow-y:scroll` property causes scroll-bars to appear always in Windows. 

## Steps to Reproduce

1. Open IE or any browsers in Windows 
2. Navigate to : https://engineering.cerner.com/terra-core/tests/terra-table/table/fill
3. Observe vertical scroll bar appears even when no content is hidden.

## Additional Context / Screenshots

<img width="1673" alt="Screen Shot 2021-07-22 at 4 45 05 PM" src="https://user-images.githubusercontent.com/12869809/126630791-2a612e93-483f-433e-83a5-b99a34af9964.png">

Same can be seen on terra-table WDIO screenshots as well : 
https://github.com/cerner/terra-core/blob/main/packages/terra-table/tests/wdio/__snapshots__/reference/terra-default-theme/en/chrome_medium/table-spec/fill.png

## Expected Behavior

scroll-bar should display only when content is overflown. 

## Possible Solution

Modifying the `overflow-y:scroll` property `overflow-y:auto` will make scroll bar to display only when it is required and removes empty scrolls on windows browsers as well as from terra-wdio-screen shots.

## Environment

* Component Name and Version:
* Browser Name and Version: Chrome Windows, IE, Chromium Windows
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile): Windows

**Comment by ryanthemanuel** [comment]
@supreethmr, this is actually by design. Parsing through old conversations with Dave Kasper:

"It's a visual bug in IE, but we cannot disable the rule as it's only this allowing the columns to align when using a fixed header implementation. All other browser reserve the inset scroll bar width, but hides the controls, IE doesn't bother to hide them"

**Comment by supreethmr** [comment]
Thank you Ryan..!! I was not aware of this and I didn't noticed column misalignment happens only in IE when table body gets scroll bar. Since row are block elements scrollbar pushes columns further to right of the scrollbar this causes misalignment with header as shown in below screenshot.

<img width="1285" alt="Screen Shot 2021-07-30 at 12 35 37 PM" src="https://user-images.githubusercontent.com/12869809/127617797-7bcea107-665f-47c4-a94f-7d0d6fd3206c.png">

**Comment by supreethmr** [comment]
> @supreethmr, this is actually by design. Parsing through old conversations with Dave Kasper:
> 
> "It's a visual bug in IE, but we cannot disable the rule as it's only this allowing the columns to align when using a fixed header implementation. All other browser reserve the inset scroll bar width, but hides the controls, IE doesn't bother to hide them"

Actually It's not just on IE, scroll bar appears on all browsers in windows machine but not on Mac.

**Comment by supreethmr** [comment]
Scroll value for property `overflow-y` will always display scrollbars whether or not any content is actually clipped. 
Mac-OS renders scrollbars inline to the div / container without affecting cells alignment and will only display the scroll when the content is clipped and mouse is hovered on table.

When Scroll appears on Windows browsers it alters cell alignment and shifts cells further to the left of the scrollbar. When the content is clipped it display active scroll-bar and when the content is not clipped it display disabled empty scroll-bar as shown in below screenshot.

<img width="1673" alt="Screen Shot 2021-07-22 at 4 45 05 PM" src="https://user-images.githubusercontent.com/12869809/128820563-f00c3093-f25c-4e29-aa5f-bc15f296521e.png">

So not sure if we have to consider fixing this visual difference between operating systems ( MAC and Windows ).

As part of investigation I found a way to keep scroll behavior same on both OS by making overflow to be hidden by default and render scroll on mouse hover. Instead of displaying disabled / empty scroll always on div / container. scroll will be added only on mouse hover so that it matches the scroll behavior of windows with macOs to some extent.


I have created draft to test the same app deployed link on Windows machine : https://github.com/cerner/terra-core/pull/3474/files#diff-7b71bcc6e5c6930d2c744330229fef9f390086cbe33a9f1a106faaf781eb3e78 

![tablescrollbar](https://user-images.githubusercontent.com/12869809/128827641-60c174fb-8fce-4249-98d7-7f813b11bd6e.gif)

Since scroll-bar on windows does alter alignment of cells we can see the movement of cells when scroll is added and removed. 

@neilpfeiffer @tbiethman please share your thoughts on considering the above stated visual difference between two OS as a Issue OR considering it as one of the differences between two OS.
